File: python_version/cli/commands/manage_structures.py
import os
from cli.utils.config_manager import ConfigManager
from cli.utils.ui import (
    print_command_banner, print_success, print_error, print_warning,
    print_info, print_section, print_list_item,
    styled_text_input, styled_select, styled_confirm,
    separator, console
)

def add_new_structure(config_manager: ConfigManager) -> bool:
    """Add a new data structure to the configuration"""
    
    name = styled_text_input("Data structure name (e.g., tree)")
    
    if not name:
        print_error("Data structure name cannot be empty!")
        return False
    
    folder = styled_text_input(
        f"Folder name (press Enter for '{name}')",
        default=name
    ) or name
    
    if config_manager.add_data_structure(name, folder):
        separator()
        console.print()
        print_success(f"Added data structure: {name} → {folder}")
        console.print()
        return True
    else:
        print_error(f"Data structure '{name}' already exists!")
        return False

def list_structures(config_manager: ConfigManager):
    """List all available data structures"""
    structures = config_manager.get_data_structures()
    
    if not structures:
        print_warning("No data structures found!")
        return
    
    print_section("Available Data Structures")
    for name, folder in structures.items():
        print_list_item(name, folder)
    console.print()

def remove_structure(config_manager: ConfigManager):
    """Remove a data structure"""
    structures = config_manager.get_data_structures()
    
    if not structures:
        print_warning("No data structures to remove!")
        return
    
    name = styled_select(
        "Select data structure to remove",
        list(structures.keys())
    )
    
    confirm = styled_confirm(
        f"Remove '{name}'?",
        default=False
    )
    
    if confirm and config_manager.remove_data_structure(name):
        separator()
        console.print()
        print_success(f"Removed data structure: {name}")
        console.print()
    else:
        print_warning("Operation cancelled")

def main(config: dict):
    """Manage data structures"""
    print_command_banner("Manage Data Structures")
    console.print()
    
    config_manager = ConfigManager()
    
    action = styled_select(
        "What would you like to do?",
        [
            "List all data structures",
            "Add new data structure",
            "Remove data structure"
        ]
    )
    
    separator()
    console.print()
    
    if "List" in action:
        list_structures(config_manager)
    elif "Add" in action:
        add_new_structure(config_manager)
    elif "Remove" in action:
        remove_structure(config_manager)

File: go-cli/internal/commands/manage_structures.go
package commands

import (
	"leetcli/internal/config"
)

func ManageStructures(args []string, cfg *config.Config, ui UI) {
	ui.WriteOutput(MsgPlain, "--- Manage Data Structures ---\n")

	action := ui.PromptSelect("What would you like to do?", []string{
		"List all data structures",
		"Add new data structure",
		"Remove data structure",
	})

	switch action {
	case "List all data structures":
		listStructures(cfg, ui)
	case "Add new data structure":
		addNewStructure(cfg, ui)
	case "Remove data structure":
		removeStructure(cfg, ui)
	}
}

func listStructures(cfg *config.Config, ui UI) {
	structures := cfg.GetDataStructures()
	if len(structures) == 0 {
		ui.WriteOutput(MsgInfo, "No data structures found!")
		return
	}
	ui.WriteOutput(MsgPlain, "Available Data Structures:")
	for name, folder := range structures {
		ui.WriteOutput(MsgPlain, "  %-20s %s", name, folder)
	}
}

func addNewStructure(cfg *config.Config, ui UI) {
	name := ui.PromptText("Data structure name (e.g., tree)")
	if name == "" {
		ui.WriteOutput(MsgError, "Data structure name cannot be empty!")
		return
	}

	folder := ui.PromptText("Folder name (press Enter for same)")
	if folder == "" {
		folder = name
	}

	if cfg.AddDataStructure(name, folder) {
		cfg.Save()
		ui.WriteOutput(MsgSuccess, "Added data structure: %s -> %s", name, folder)
	} else {
		ui.WriteOutput(MsgError, "Data structure '%s' already exists!", name)
	}
}

func removeStructure(cfg *config.Config, ui UI) {
	structures := cfg.GetDataStructures()
	if len(structures) == 0 {
		ui.WriteOutput(MsgInfo, "No data structures to remove!")
		return
	}

	names := make([]string, 0, len(structures))
	for n := range structures {
		names = append(names, n)
	}

	name := ui.PromptSelect("Select data structure to remove", names)
	if name == "" {
		return
	}

	confirm := ui.PromptConfirm("Remove '" + name + "'?")
	if !confirm {
		ui.WriteOutput(MsgInfo, "Operation cancelled")
		return
	}

	if cfg.RemoveDataStructure(name) {
		cfg.Save()
		ui.WriteOutput(MsgSuccess, "Removed data structure: %s", name)
	} else {
		ui.WriteOutput(MsgError, "Data structure '%s' not found!", name)
	}
}
